### PR TITLE
Add transform function

### PR DESCRIPTION
**Pull request by sobychacko**

### CODE REVIEW

File: function/transformer-function/pom.xml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    </parent>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-integration</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-messaging</artifactId>
```
Comment (artembilan) [inline function/transformer-function/pom.xml]: this one comes as transitive from `spring-integration-core`.

File: function/transformer-function/pom.xml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter</artifactId>
```
Comment (artembilan) [inline function/transformer-function/pom.xml]: This one comes as transitive from `spring-boot-starter-integration`

File: function/transformer-function/src/main/java/io/pivotal/java/function/transformer/function/TransformerFunctionApplication.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2020-2020 the original author or authors.
```
Comment (artembilan) [inline function/transformer-function/src/main/java/io/pivotal/java/function/transformer/function/TransformerFunctionApplication.java]: Looks suspicious.
Why not just one `2020`?

File: function/transformer-function/src/main/java/io/pivotal/java/function/transformer/function/TransformerFunctionConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+							headers.containsKey(MessageHeaders.CONTENT_TYPE)
+									? headers.get(MessageHeaders.CONTENT_TYPE).toString()
+									: "application/json";
+					if (contentType.contains("text") || contentType.contains("json") || contentType.contains("x-spring-tuple")) {
+						message = MessageBuilder.withPayload(new String(((byte[]) message.getPayload())))
+								.copyHeaders(message.getHeaders())
+								.build();
+					}
+				}
+				final ExpressionEvaluatingTransformer expressionEvaluatingTransformer = new ExpressionEvaluatingTransformer(
```
Comment (artembilan) [inline function/transformer-function/src/main/java/io/pivotal/java/function/transformer/function/TransformerFunctionConfiguration.java]: This is wrong: you are  going to create an `ExpressionEvaluatingTransformer` instance for every single request message.
This has to be as a bean any way. There are many things done by the framework when a bean is initialized.

File: function/transformer-function/pom.xml (hunk 1 above)
Comment (sobychacko) [reply function/transformer-function/pom.xml]: Makes sense about the naming. I will give it a thought.

Comment (sobychacko) [reply function/transformer-function/pom.xml]: spel-function sounds good to me.

File: function/spel-function/src/main/java/io/pivotal/java/function/spel/function/SpelFunctionConfiguration.java [hunk 5]
```diff
@@ -28,11 +28,11 @@
...
 
 @Configuration
-@EnableConfigurationProperties(TransformerFunctionProperties.class)
-public class TransformerFunctionConfiguration {
+@EnableConfigurationProperties(SpelFunctionProperties.class)
+public class SpelFunctionConfiguration {
 
 	@Bean
-	public Function<Message<?>, Message<?>> transformFunction(TransformerFunctionProperties transformerFunctionProperties) {
+	public Function<Message<?>, Message<?>> transformerFunction(ExpressionEvaluatingTransformer expressionEvaluatingTransformer) {
```
Comment (artembilan) [inline function/spel-function/src/main/java/io/pivotal/java/function/spel/function/SpelFunctionConfiguration.java]: I think `spelFunction` is right bean name...

File: function/spel-function/src/main/java/io/pivotal/java/function/spel/function/SpelFunctionConfiguration.java [hunk 6]
```diff
@@ -46,10 +46,15 @@
...
 				}
-				final ExpressionEvaluatingTransformer expressionEvaluatingTransformer = new ExpressionEvaluatingTransformer(
-						new SpelExpressionParser()
-								.parseExpression(transformerFunctionProperties.getExpression()));
 				return expressionEvaluatingTransformer.transform(message);
 			};
 	}
+
+	@Bean
+	public ExpressionEvaluatingTransformer expressionEvaluatingTransformer(SpelFunctionProperties spelFunctionProperties){
```
Comment (artembilan) [inline function/spel-function/src/main/java/io/pivotal/java/function/spel/function/SpelFunctionConfiguration.java]: White space before curly brace.

**Comment by artembilan** [comment]
Merged as https://github.com/pivotal/java-functions/commit/4c58192cb5c21ae3f14337b1f66399a77dff0b35 after some clean up.

**Pull request closed without merging**